src/NuGet.Core/NuGet.Protocol/Resources/RepositorySignatureResource.cs

```diff
@@ -35,5 +38,11 @@ public RepositorySignatureResource(bool allRepositorySigned, IEnumerable<IReposi
             RepositoryCertificateInfos = repositoryCertInfos;
         }
 
+        public void UpdateRepositorySignatureInfo()
+        {
+            RepositorySignatureInfoProvider.Instance.AddRepositorySignatureInfo
```

Comment:
Yuck.  Instead of static, how about this?

1.  Register `RepositorySignatureInfoProvider` as a resource provider.
2.  In `RepositorySignatureResourceProvider` fetch a `RepositorySignatureInfoProvider` instance.
3.  Pass the instance to the `RepositorySignatureResource` constructor.